src/libsyntax/parse/parser.rs

```diff
@@ -1576,7 +1577,7 @@ impl<'a> Parser<'a> {
             let ident = self.parse_ident()?;
             let mut generics = self.parse_generics()?;
 
-            let d = self.parse_fn_decl_with_self(|p: &mut Parser<'a>| {
+            let mut d = self.parse_fn_decl_with_self(|p: &mut Parser<'a>| {
```

Comment:
Preexisting, but `d` is a pretty bad binding name.

Reply:
Done.